phoenix-core/src/main/java/org/apache/phoenix/expression/function/RoundWeekExpression.java

```diff
@@ -17,25 +17,36 @@
  */
 package org.apache.phoenix.expression.function;
 
+import java.util.Calendar;
+import java.util.Date;
 import java.util.List;
 
 import org.apache.phoenix.expression.Expression;
-import org.joda.time.DateTime;
+
+import static java.lang.Math.abs;
 
 /**
  * 
- * Rounds off the given {@link DateTime} to the nearest Monday.
+ * Rounds off the given {@link Date} to the nearest Monday.
  */
-public class RoundWeekExpression extends RoundJodaDateExpression {
+public class RoundWeekExpression extends RoundJavaDateExpression {
 
-    public RoundWeekExpression(){}
+    public RoundWeekExpression() {}
     
     public RoundWeekExpression(List<Expression> children) {
        super(children);
     }
 
     @Override
-    public long roundDateTime(DateTime dateTime) {
-       return dateTime.weekOfWeekyear().roundHalfEvenCopy().getMillis();
+    public long roundDateTime(Date dateTime) {
+        long nextMonday = new CeilWeekExpression().roundDateTime(dateTime);
+        long prevMonday = new FloorWeekExpression().roundDateTime(dateTime);
```

Comment:
Likewise, we can't use DateUtils.round here?

Reply:
I will check it again.

Reply:
DateUtils, doesn't support rounding to weeks.